include/gul14/traits.h

```diff
@@ -1,29 +1,22 @@
 /**
  * \file    traits.h
  * \brief   Some metaprogramming traits for the General Utility Library.
- * \authors \ref contributors
+ * \authors \ref contributors, Michael Park
  * \date    Created on 20 August 2021
  *
- * \copyright Copyright 2021-2023 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ * Copyright 2015-2017 Michael Park (implementations of invoke, invoke_result,
+ * invoke_result_t, is_invocable, is_invocable_r), copyright 2021-2024 Deutsches
+ * Elektronen-Synchrotron (DESY), Hamburg (other implementations and modifications)
  *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Lesser General Public License as published
- * by the Free Software Foundation, either version 2.1 of the license, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ * Distributed under the Boost Software License, Version 1.0.
+ * (See \ref license_boost_1_0 or http://boost.org/LICENSE_1_0.txt)
```

Comment:
I'm not so sure we can or should change existing code's license (from LGPL to Boost). At least we would need the explicit consent of all authors for the change.

This needs to be also documented in `debian/copyright`
* License change
* Author change (addition)

Reply:
I changed the entire header to Boost to avoid having two licenses in a single source file. All other authors are employed by DESY, so strictly speaking we would need a management decision. But since that won't happen I'd vote for "just do it", it is just a few lines of code after all.

`debian/copyright`: Yes, of course, I keep forgetting about that.

Reply:
Several ;-)

Who invented the `BOOST-1` tag? ;-D

![image](https://github.com/gul-cpp/gul14/assets/16012374/5eec9e58-0114-416b-be1e-d9cf37890b2a)

Links
* https://wiki.debian.org/CopyrightReviewTools
* https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
* https://spdx.org/licenses/

![image](https://github.com/gul-cpp/gul14/assets/16012374/30f7b5ad-fc60-46df-9768-6927e5d62c81)

\*hust\*

![image](https://github.com/gul-cpp/gul14/assets/16012374/405de162-e175-49b3-ad73-58465bab82a0)

Reply:
```sh
meson build --prefix=/usr
cp build/debian/copyright debian/copyright
rm -Rf build
debmake -kk
```

Reply:
I have corrected BOOST-1 -> BSL-1.0. The rest seems to be OK.